portfolio/src/main/java/com/google/sps/servlets/GenreServlet.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/genre-data")
+public class GenreServlet extends HttpServlet {
+
+  private Map<String, Integer> genreVotes = new HashMap<>();
```

Comment:
If this is only being used for `doPost()` can we make it a local variable, instead of a class variable?

Reply:
It is actually also used in the doGet() function, with the toJson(genreVotes) conversion. This is why I initially had it as a class variable.

Reply:
Ah hmm... how long do servlets stay in memory? `doGet()` doesn't fetch from the database, so won't it always be empty on the first GET, and then only ever contain things if someone has recently done a POST?

Reply:
That is true. I was wondering why the votes weren't staying after refreshing the page. I have to resolve this.